src/gui/properties/speedplotview.cpp

```diff
@@ -33,15 +33,38 @@
 #include "base/global.h"
 #include "base/utils/misc.h"
 
+SpeedPlotView::Averager::Averager(int _divider, boost::circular_buffer<PointData> &_sink)
+    : cnt(0)
+    , divider(_divider)
+    , sink(_sink)
+{
+    acc.reset();
+}
+inline void SpeedPlotView::Averager::push(const PointData &p)
+{
+    acc.x += p.x;
+    for (int id = UP; id < NB_GRAPHS; ++id)
+        acc.y[id] += p.y[id];
+    cnt = (cnt + 1) % divider;
+    if (cnt == 0) {
+        for (int id = UP; id < NB_GRAPHS; ++id)
+            acc.y[id] /= divider;
+        acc.x /= divider;
+        // now flush out averaged data
+        sink.push_back(acc);
+        acc.reset();
```

Comment:
I think you could do `acc = {};` instead and drop reset().
But wait for others to confirm.